pkg/operator/ceph/object/objectstore.go

```diff
@@ -99,59 +105,79 @@ func reconcileRealm(context *Context, serviceIP string, port int32) error {
+			// need to change this logging
+			return errors.Wrapf(err, "failed to add object store %s in rgw zone %s in zonegroup %s in realm %s", context.Name, zoneID, zoneGroupID, realmID)
+		}
+	} else {
+		// create the realm if it doesn't exist yet
+		output, err := runAdminCommand(context, "realm", "get")
+		if err != nil {
+			updatePeriod = true
+			output, err = runAdminCommand(context, "realm", "create", defaultArg)
+			if err != nil {
+				return errors.Wrapf(err, "failed to create rgw realm %s", context.Name)
+			}
 		}
-	}
 
-	realmID, err := decodeID(output)
-	if err != nil {
-		return errors.Wrap(err, "failed to parse realm id")
-	}
+		realmID, err := decodeID(output)
+		if err != nil {
+			return errors.Wrapf(err, "failed to parse realm id")
+		}
 
-	// create the zonegroup if it doesn't exist yet
-	output, err = runAdminCommand(context, "zonegroup", "get")
-	if err != nil {
-		updatePeriod = true
-		output, err = runAdminCommand(context, "zonegroup", "create", "--master", endpointArg, defaultArg)
+		// create the zonegroup if it doesn't exist yet
+		output, err = runAdminCommand(context, "zonegroup", "get")
+		if err != nil {
+			updatePeriod = true
+			output, err = runAdminCommand(context, "zonegroup", "create", "--master", endpointArg, defaultArg)
+			if err != nil {
+				return errors.Wrapf(err, "failed to create rgw zonegroup for %s", context.Name)
+			}
+		}
+
+		zoneGroupID, err := decodeID(output)
 		if err != nil {
-			return errors.Wrapf(err, "failed to create rgw zonegroup for %s", context.Name)
+			return errors.Wrapf(err, "failed to parse zone group id")
 		}
-	}
 
-	zoneGroupID, err := decodeID(output)
-	if err != nil {
-		return errors.Wrap(err, "failed to parse zone group id")
-	}
+		// create the zone if it doesn't exist yet
+		output, err = runAdminCommand(context, "zone", "get", zoneArg)
+		if err != nil {
+			updatePeriod = true
+			output, err = runAdminCommand(context, "zone", "create", "--master", endpointArg, zoneArg, defaultArg)
+			if err != nil {
+				return errors.Wrapf(err, "failed to create rgw zonegroup for %s", context.Name)
+			}
+		}
 
-	// create the zone if it doesn't exist yet
-	output, err = runAdminCommand(context, "zone", "get", zoneArg)
-	if err != nil {
-		updatePeriod = true
-		output, err = runAdminCommand(context, "zone", "create", "--master", endpointArg, zoneArg, defaultArg)
+		zoneID, err := decodeID(output)
 		if err != nil {
-			return errors.Wrapf(err, "failed to create rgw zonegroup for %s", context.Name)
+			return errors.Wrapf(err, "failed to parse zone id")
 		}
-	}
-	zoneID, err := decodeID(output)
-	if err != nil {
-		return errors.Wrap(err, "failed to parse zone id")
+
+		logger.Infof("RGW: realm=%s, zonegroup=%s, zone=%s", realmID, zoneGroupID, zoneID)
 	}
 
 	if updatePeriod {
 		// the period will help notify other zones of changes if there are multi-zones
-		_, err := runAdminCommandNoRealm(context, "period", "update", "--commit")
```

Comment:
If we only update the period when the object store is created, what happens if the operator stops or restarts between the time that a realm/zone/zonegroup is created and the time when the object store is created? Does it just mean other clusters won't be notified of the change until the period is committed?